atlp-rwanda/stackup2-barefoot-backend
# #171380746 User descriptive signup validation errors

**What does this PR do?**
for descriptive user signup validation errors

**Description of Task to be completed?**
add src/middlewares/authValidation.js middleware
add src/middlewares/validation.js service
modify and add if email exist script in src/controllers/authentication.controller.js
add test script in src/controllers/authentication.test.js

**How should this be manually tested?**
- clone the project
- open your project in cmd, and then write _npm install_ to install packages
- while signing up try to write invalid text on email for instance.
- run the project by writing in cmd _npm start_ . 
- test signup endpoint in postman for instance, and see the output. Actually will return error because email is invalid as aforementioned.

**Any background context you want to provide?**
N/A

**What are the relevant pivotal tracker stories?**
[#171380746](https://www.pivotaltracker.com/n/projects/2435482/stories/171380746)
